L1TriggerScouting/OnlineProcessing/plugins/BMTFStubMultiBxSelector.cc

```diff
@@ -0,0 +1,167 @@
+          uniqueStubSelectedBxs->insert(bx - i);
+      }
+    }
+    // Wheel condition: enough longitudinally "neighbouring" stubs
+    else if (condition_ == l1ScoutingRun3::Wheel) {
+      for (unsigned i = 0; i < std::min(bxWindowLength_, bx); ++i) {
+        // Prepare pattern and add to window vector
+        vWheelPatternBx[i] = makeWheelPattern(stubsCollection, bx - i);
+      }
+
+      // Check if there are stubs in different BXs with neighbouring wheels
+      // For example (with window of 3 BXs, neighbouring condition abs(wheel_pair) <= 1)
+      // BX-2         BX-1        BX0
+      // s0(wh = 2)   s0(wh=0)    s1(wh=-2)
+      // s1(wh = 1)
+      //
+      // => valid window! (neighbouring stubs s1 in BX-2 and s0 in BX-1, assuming nStub threshold is satisfied)
+      bool validWindow = windowHasCloseWheels(vWheelPatternBx) && (numStubsWindow >= minNBMTFStub_);
+
+      // If window is valid....
+      if (validWindow) {
+        // ...loop over window to add BXs (std::min to include edge case of first BXs)
+        for (unsigned i = 0; i < std::min(bxWindowLength_, bx); ++i)
+          uniqueStubSelectedBxs->insert(bx - i);
+      }
+    }
+  }  // end orbit loop
+
+  // Convert set of selected BXs to a vector and put collection in event content
+  std::unique_ptr<std::vector<unsigned>> stubSelectedBx =
+      std::make_unique<std::vector<unsigned>>(uniqueStubSelectedBxs->begin(), uniqueStubSelectedBxs->end());
+  iEvent.put(std::move(stubSelectedBx), "SelBx");
+}
+
+unsigned BMTFStubMultiBxSelector::makeWheelPattern(const edm::Handle<OrbitCollection<l1ScoutingRun3::BMTFStub>>& stubs,
+                                                   unsigned bx) {
+  // 5 wheel numbers + 2 to handle boundaries in a more comfortable way:
+  // 0b         0   X   X   X   X   X   0
+  // wheels   bnd  +2  +1   0  -1  -2 bnd
+  // position   6   5   4   3   2   1   0
+  unsigned wheelPatternBx = 0;
+  for (const auto& s : stubs->bxIterator(bx))
+    wheelPatternBx |= (1 << (s.wheel() + 3));
+  return wheelPatternBx;
+}
+
+bool BMTFStubMultiBxSelector::windowHasCloseWheels(const std::vector<unsigned>& bxs) {
+  for (size_t i = 0; i < std::size(bxs); ++i) {
+    for (size_t j = i + 1; j < std::size(bxs); ++j) {
+      // Assume for example that we have the following patterns
+      //          bwwwwwb (b = boundary, w = wheel fields)
+      // BX-2 : 0b0010000 (stub with wheel=+2)
+      // BX-1 : 0b0001000 (stub with wheel=+1)
+      // Bitwise AND comparison between:
+      // BX-2 : 0b0010000
+      // BX-1 : 0b0011100
+      //            bsb   (s = stub wheel, b = boundary wheels)
+      unsigned compare = bxs[i] & ((bxs[j] << 1) | bxs[j] | (bxs[j] >> 1));
+      bool checkWindow = compare & 0b0111110;
+      if (checkWindow)
+        return true;
+    }
+  }
+  return false;
+}
+
+void BMTFStubMultiBxSelector::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
```

Comment:
Probably it is better to set a proper description with the parameter sets expected, and default values